### PR TITLE
ARC-134,135,136: Grouping Metadata, Add Description, & Populate Date Added

### PR DESCRIPTION
**Pull request by y0ungy3**

### CODE REVIEW

File: src/state/upload/upload-reducers.js [hunk 1]
```diff
@@ -37,9 +38,24 @@ export default function (state = initialState, action) {
 
         case uploadActionTypes.FETCH_ITEMS_SUCCEEDED: {
             const { groups } = action.data;
+            let ID = null;
```
Comment (mr2ert) [inline src/state/upload/upload-reducers.js]: Why did you have to change this reducer?

Comment (y0ungy3) [reply src/state/upload/upload-reducers.js]: it's basically to show the meta data for generic on page load without having to click the checkbox

Comment (mr2ert) [reply src/state/upload/upload-reducers.js]: Ah, I missed that this was for the upload screen only -- thought it was also potentially affecting the sidebar.

File: src/state/upload/upload-reducers.js [hunk 2]
```diff
@@ -37,9 +38,24 @@ export default function (state = initialState, action) {
...
+                    ID = groups[i].id;
+                }
+            }
+            let array = [];
+            array = array.concat(ID);
             return {
                 ...state,
-                groups: groups.map((object) => { const obj = object; obj.checkbox = false; return obj; }),
+                groups: groups.map((object) => {
+                    const obj = object;
```
Comment (mr2ert) [inline src/state/upload/upload-reducers.js]: Why do we need to do this?  Can't we just do `object.checkbox = ...`?

File: src/state/upload/upload-reducers.js [hunk 3]
```diff
@@ -37,9 +38,24 @@ export default function (state = initialState, action) {
...
+                if (groups[i].name === 'Generic') {
+                    ID = groups[i].id;
+                }
+            }
+            let array = [];
+            array = array.concat(ID);
             return {
                 ...state,
-                groups: groups.map((object) => { const obj = object; obj.checkbox = false; return obj; }),
+                groups: groups.map((object) => {
```
Comment (mr2ert) [inline src/state/upload/upload-reducers.js]: I'd prefer either `group` or `g` as the name.

File: src/state/upload/upload-reducers.js [hunk 4]
```diff
@@ -37,9 +38,24 @@ export default function (state = initialState, action) {
...
+                    const obj = object;
+                    if (object.name === 'Generic') { obj.checkbox = true; } else {
```
Comment (mr2ert) [inline src/state/upload/upload-reducers.js]: The makes the for-loop above unneeded.  Set `array` equal to a https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set, add the ids here, and use http://stackoverflow.com/a/20070691/2336149 when setting it to `allItemID` below.

**Review by mr2ert — CHANGES_REQUESTED** [review]
There are a couple of style things as well that we can worry about next quarter.

File: src/state/upload/upload-reducers.js (hunk 2 above)
Comment (y0ungy3) [reply src/state/upload/upload-reducers.js]: tried that before, eslint doesn't let you...

File: src/state/upload/upload-reducers.js (hunk 4 above)
Comment (y0ungy3) [reply src/state/upload/upload-reducers.js]: feel free to make changes

File: src/state/upload/upload-reducers.js (hunk 3 above)
Comment (y0ungy3) [reply src/state/upload/upload-reducers.js]: it's named "groups" because that's what the JSON object in the backend is named...so it'll be consistent

Comment (mr2ert) [reply src/state/upload/upload-reducers.js]: I meant for `object`

**Review by mr2ert — APPROVED** [review]

**Pull request merged**